### DISCUSSION TITLE
CLI Question

### DISCUSSION THREAD

**Discussion by micogg** [immich-app/immich#819] [Q&A]
I'm running Immich in a Truenas Scale Docker/App.

I would like to transfer ~1,000 from a separate NextCloud App.  
I found the CLI option, but I'm a little lost on how to use it. 
Does the CLI command have to be run within the Immich docker or wherever the image source files are being stored?
Additionally, the first step of the ReadMe states to run npm i -g immich, but Scale does not support NPM to my knowledge.
Is the NPM Immich installing the Immich CLI dependencies, or should they already be installed with the main Immich install in the docker?

**Comment by alextran1502** [comment]
Do you have access to the terminal where the Nextcloud app stores the files?

From the terminal, you can install Node.js and then install Immich CLI, which is a Node's package that can run as a standalone binary.

You can then use the CLI command to upload all the files in the Nextcloud app to an Immich server.

If you need additional help, please let me know.

**Reply by micogg** [reply]
Thanks! Truenas won't allow you to install NPM so I made the folder a SMB share, downloaded them to a windows computer and ran the CLI command.  All worked well.

A separate but somewhat related question, there is an option on the docker build for custom/additional storage. 
If I attach an additional storage location, will Immich automatically pull files from it? 
I may have missed it, but I did not see an auto-backup option for PC or a shared drive.

**Reply by alextran1502** [reply]
The docker build basically offer a more convenient way to run the CLI without installing the Npm and Nodejs. There is currently no "watching" option for PC or Server yet. I am working on a prototype version of CLI that can eventually do that though
